Stripe/STPPushProvisioningContext.m

```diff
@@ -0,0 +1,95 @@
+//
+//  STPPushProvisioningContext.m
+//  Stripe
+//
+//  Created by Jack Flintermann on 9/27/18.
+//  Copyright © 2018 Stripe, Inc. All rights reserved.
+//
+
+#import "STPPushProvisioningContext.h"
+#import "STPEphemeralKeyManager.h"
+#import "STPAPIClient+Private.h"
+#import "STPAPIClient+PushProvisioning.m"
+#import "STPEphemeralKey.h"
+
+@interface STPPushProvisioningContext()
+@property (nonatomic, strong) STPEphemeralKeyManager *keyManager;
+@property (nonatomic, strong, nullable) STPEphemeralKey *ephemeralKey;
+@end
+
+@implementation STPPushProvisioningContext
+
+- (instancetype)initWithKeyProvider:(id<STPIssuingCardEphemeralKeyProvider>)keyProvider {
+    self = [super init];
+    if (self) {
+        _keyManager = [[STPEphemeralKeyManager alloc] initWithKeyProvider:keyProvider apiVersion:[STPAPIClient apiVersion] performsEagerFetching:NO];
+    }
+    return self;
+}
+
+- (void)generateRequestConfiguration:(nullable STPPushProvisioningRequestConfigurationBlock)completion {
+        
+        //        stpDispatchToMainThreadIfNecessary(^{
+        //            completion(self.customer, nil);
+        //        });
+}
+
+- (void)_generateRequestConfiguration:(STPPushProvisioningRequestConfigurationBlock)completion {
+    [self.keyManager getOrCreateKey:^(STPEphemeralKey * _Nullable ephemeralKey, NSError * _Nullable keyError) {
+        if (keyError != nil) {
+            completion(nil, keyError);
+            return;
+        }
+        
+        STPAPIClient *client = [STPAPIClient apiClientWithEphemeralKey:ephemeralKey];
+        [client retrieveIssuingCardWithID:ephemeralKey.issuingCardID completion:^(STPIssuingCard * _Nullable card, NSError * _Nullable cardError) {
+            if (cardError != nil) {
+                completion(nil, cardError);
+                return;
+            }
+            
+            if (![[PKPassLibrary new] canAddPaymentPassWithPrimaryAccountIdentifier:card.cardId]) {
+                NSError *error = nil; // TODO
+                completion(nil, error);
+            }
+            
+            PKAddPaymentPassRequestConfiguration *configuration = [[PKAddPaymentPassRequestConfiguration alloc] initWithEncryptionScheme:PKEncryptionSchemeECC_V2];
+            configuration.cardholderName = card.name;
+            configuration.primaryAccountSuffix = card.last4;
+            if (@available(iOS 12.0, *)) {
+                configuration.style = PKAddPaymentPassStylePayment;
```

Comment:
Offhand I don't remember the incantation necessary, but we're still supporting Xcode 9 and it doesn't know about this symbol.

I don't actually know what that'll mean for that static binary that we provide via github. Instructions are to generate it with the oldest supported major version of Xcode, which means this would only be set for users who integrate via one of the other methods...